hack/generate/csv.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+set -Eeuo pipefail
+
+template="${1:?Provide template file as arg[1]}"
+target="${2:?Provide a target CSV file as arg[2]}"
+
+# shellcheck disable=SC1091,SC1090
+source "$(dirname "${BASH_SOURCE[0]}")/../lib/metadata.bash"
+
+registry="registry.svc.ci.openshift.org/openshift"
+serving="${registry}/knative-v$(metadata.get components.serving):knative-serving"
+eventing="${registry}/knative-v$(metadata.get components.eventing):knative-eventing"
+
+declare -A images
+images["queue-proxy"]="${serving}-queue"
+images["activator"]="${serving}-activator"
+images["autoscaler"]="${serving}-autoscaler"
+images["autoscaler-hpa"]="${serving}-autoscaler-hpa"
+images["controller"]="${serving}-controller"
+images["webhook"]="${serving}-webhook"
+images["storage-version-migration-serving-$(metadata.get components.serving)__migrate"]="${serving}-storage-version-migration"
+
+images["3scale-kourier-gateway"]="docker.io/maistra/proxyv2-ubi8:$(metadata.get components.kourier)"
+images["3scale-kourier-control"]="${registry}/knative-v$(metadata.get components.serving):kourier"
```

Comment:
Regarding the other comment: This is wrong. The serving version does not necessarily equal the kourier version, hence we need a kourier component in the metadata too.

Reply:
Done.